src/Http/Authentication.Core/src/AuthenticationService.cs

```diff
@@ -77,11 +104,13 @@ public virtual async Task<AuthenticateResult> AuthenticateAsync(HttpContext cont
         // Handlers should not return null, but we'll be tolerant of null values for legacy reasons.
         var result = (await handler.AuthenticateAsync()) ?? AuthenticateResult.NoResult();
 
+        _metrics?.AuthenticatedRequest(scheme, result);
```

Comment:
`AuthenticationService.AuthenticateAsync` can throw errors:
* No auth scheme
* Missing handler
* Can handler AuthenticateAsync throw?
* Can transform TransformAsync throw?

This metric should handle error cases.

Also, do people case how long authenticating takes? Should this be a histogram rather than a counter? A histogram can be used for both measuring count and duration.

Reply:
Recent changes report exceptions via the `error.type` tag, and `AuthenticateAsync()` is now reported as a histogram.